examples/task_controller_client/section_control_implement_sim.cpp

```diff
@@ -0,0 +1,318 @@
+		connector->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::ConnectorYOffset));
+		connector->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::ConnectorType));
+
+		boom->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::BoomXOffset));
+		boom->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::BoomYOffset));
+		boom->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::BoomZOffset));
+		boom->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::ActualWorkingWidth));
+		boom->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::ActualCondensedWorkingState1To16));
+		boom->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::SetpointCondensedWorkingState1To16));
+		boom->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::SectionControlState));
+
+		for (std::size_t i = 0; i < NUMBER_SECTIONS; i++)
+		{
+			boom->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::Section1) + i);
+		}
+
+		product->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::TankCapacity));
+		product->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::TankVolume));
+		product->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::LifetimeApplicationVolumeTotal));
+		product->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::PrescriptionControlState));
+		product->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::ActualCulturalPractice));
+		product->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::TargetRate));
+		product->add_reference_to_child_object(static_cast<std::uint16_t>(ImplementDDOPObjectIDs::ActualRate));
+	}
+	return retVal;
+}
+
+bool SectionControlImplementSimulator::request_value_command_callback(std::uint16_t,
+                                                                      std::uint16_t DDI,
+                                                                      std::uint32_t &value,
+                                                                      void *parentPointer)
+{
+	if (nullptr != parentPointer)
+	{
+		auto sim = reinterpret_cast<SectionControlImplementSimulator *>(parentPointer);
+		switch (DDI)
+		{
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::MaximumVolumeContent):
+			{
+				value = 4542494; // 1200 Gallons in ml
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::ActualVolumeContent):
+			{
+				value = 3785000; // 1000 Gal in ml;
+			}
+			break;
+
+			case static_cast<std::uint16_t>(isobus::DataDescriptionIndex::SectionControlState):
+			{
+				value = 1; // AUTO mode
```

Comment:
Not sure what this does, but could this be related to `sim->get_section_control_state`?